### ISSUE TITLE
Components should output all the geometry options after orientation study

### ISSUE THREAD

**Issue by mostaphaRoudsari** [mostaphaRoudsari/ladybug#169]
See this comment: http://www.grasshopper3d.com/xn/detail/2985220:Comment:1347207

**Comment by chriswmackey** [comment]
@mostaphaRoudsari ,
I was just thinking this past week that this would be so helpful.  Looking forward to it!
-Chris

**Comment by ParametricMonkey** [comment]
Hi Mostapha

The other thing that I think would be useful was if there were 2 context inputs. Often I use the sunlight component to evaluate hrs of sun on windows. So when it comes to an orientation study I would like the context (neighbouring buildings) to be fixed rotation but the other context (slabs, walls, etc) of the proposed building to rotate with the windows during the orientation study. Can you add this to the wish list. Thanks

**Comment by mostaphaRoudsari** [comment]
Hi Paul, It's actually already possible. you can use parameters to add geometries which will stay at the same position during the study. I will upload an example file soon.

**Comment by ayezioro** [comment]
Hi,
This was one of my first requests at the LB starting release days. Wasn't aware it is already working. So thanks, Mostapha.
This is an example i've just tried with the rotated context:
https://www.dropbox.com/s/fvjhpd9axny3kpj/orientationStudy_SampleFile%2BTwoContexts.gh?dl=0
Mostapha: Can you check i set the thigs right?
A new wish as a result: Baking the rotated context. I can think this can be very desirable on many situations.
-A.

**Comment by ParametricMonkey** [comment]
Hi Mostapha
I had another look at the orientation study and can't see how this is possible. When you say use parameters, do you mean the 'orientation study parameter'. 

Also, did you have any success with this request:

Is there anyway to get all the results without having to bake them? I know its possible to use totalSunlightHours to see the results but the analysisMesh only has a single geometric output. It would simplify workflow if you didn't need to bake it to see all the results. It would also be cool to get a list of the rotations as an output rather than having to calculate them separately.

Thanks. P

**Comment by ParametricMonkey** [comment]
Hi guys. Any luck with this? And Mostapha can you upload that example pls

**Comment by mostaphaRoudsari** [comment]
Hi Paul, I didn't have a chance to work on this but I'll try to put an example file for you now and post it here.

Meanwhile as you know I'm re-writing the sunlight hours analysis component and I actually took the orientation study out of the component assuming user can rotate the geometries as they want and instead I'm adding a new component that collects the geometries and the results and dump them into a folder which you can later/or while the analysis is running load them back in.

The idea initially started after I couldn't get it to work yet in Dynamo because of memory issues but then I realized it's actually very clean solution and can work for any other types of study including preparing materials for [DesignExplorer](http://tt-acm.github.io/DesignExplorer/). I don't have a working prototype right now which could have helped so much but I hope the concept is clear? Does it sound good to you?

**Comment by mostaphaRoudsari** [comment]
@ParametricMonkey [here is the example file](https://www.dropbox.com/s/anlj9313nypeeqa/orientationStudy_2.gh?dl=0). The shading rotates with the geometry while the context building stands in the same position during the run.

**Comment by antonszilasi** [comment]
@mostaphaRoudsari  is this still open? This is something I would like to do, you can assign me to the task if it is still needed.

**Comment by mostaphaRoudsari** [comment]
@antonszilasi yes. It is still open. If you're willing to give it a try please go ahead. I should just give you heads up that the code base is a mess! My apologies in advance.

**Comment by ayezioro** [comment]
Looking forward for this one.
Thanks Anton.
-A.
